exercises/concept/power-of-troy/.docs/introduction.md

```diff
@@ -0,0 +1,98 @@
+
+When creating a `std::unique_ptr`, it's preferable to use `std::make_unique()` instead of directly using `new` to allocate memory.
+`std::make_unique()` provides several advantages:
+1. **Exception Safety**: `std::make_unique()` guarantees exception safety.
+  If an exception is thrown during the construction of the object, memory will be automatically deallocated, preventing memory leaks.
+2. **Clarity**: Using `std::make_unique()` makes code clearer and more concise.
+  It eliminates the need to explicitly specify the type being allocated, as the template arguments are deduced automatically.
+3. **Optimization Opportunities**: Compilers have the opportunity to optimize `std::make_unique()` more effectively than manually allocating memory with `new`, potentially resulting in improved performance.
+4. **Avoiding Misuse**: Deleting the underlying resource is possible, when the `std::unique_ptr` is constructed manually.
+  That would lead to undefined behavior, when the `std::unique_ptr` tries to delete it at its end of scope.
+
+## Shared Pointers
+
+`std::shared_ptr` is a smart pointer that allows multiple `std::shared_ptr` objects to share ownership of the same resource.
+It keeps track of how many shared pointers are referencing the resource, and deallocates the memory only when the last shared pointer owning the resource goes out of scope or is reset.
+
+```cpp
+// Declaring and defining a shared pointer to a dynamically allocated string
+auto martian_congressional_republic = std::make_shared<std::string>("protomolecule");
+
+// Creating more shared pointer that shares ownership
+auto outer_planets_alliance = martian_congressional_republic;
+auto united_nations = martian_congressional_republic;
+```
+
+~~~~exercism/caution
+In C++17 and below, using `std::shared_ptr` with arrays via `std::make_shared<T[]>` is not directly supported.
+While it's possible to allocate arrays with `std::make_shared<T[]>`, creating shared pointers directly from them may lead to undefined behavior due to differences in memory management between single objects and arrays.
+Instead, consider using `std::vector` or custom deletion functions to manage arrays with shared pointers effectively.
+Always ensure compatibility with your compiler and standard library implementation when dealing with array allocations and shared pointers in C++17.
+~~~~
+
+## Advantages of `std::make_shared()`
+
+Similar to `std::make_unique()`, `std::make_shared()` offers benefits such as improved memory efficiency, exception safety, and readability.
+It combines memory allocation for the control block and the managed object into a single operation, enhancing efficiency and reducing the risk of memory leaks.
+Additionally, automatic deduction of template arguments simplifies code and enhances readability.
+Using `std::make_shared()` promotes cleaner, safer, and more efficient code when working with `std::shared_ptr` objects in C++.
+
+
+~~~~exercism/advanced
```

Comment:
Should I take this out, even though it is in the connected concept?

Reply:
Are students required to use `std::weak_ptr` it in the exercise? If not, I'd omit it.